pychiquito/dsl.py

```diff
@@ -87,37 +163,33 @@ class StepTypeMode(Enum):
     WG = 2
 
 
-class StepTypeContext:
-
-    def __init__(self: StepTypeContext, circuit, step_type_name: str, ):
-        self.step_type = StepType.new(step_type_name)
+class StepType:
+    def __init__(self: StepType, circuit: Circuit, step_type_name: str):
+        self.step_type = ASTStepType.new(step_type_name)
         self.circuit = circuit
+        self.step_instance = StepInstance.new(self.step_type.id)
         self.mode = StepTypeMode.SETUP
         self.setup()
-        self.mode = StepTypeMode.NoMode
+        self.mode = StepTypeMode.WG
```

Comment:
@qwang98 we are not in WG mode after setup, like with trace, there should be a function that wraps wg and puts the mode and disables it. Something like get_assignments

Reply:
Created `gen_step_instance`, which has a parallel design as `gen_witness`.

`gen_step_instance` sets the mode, creates an empty StepInstance, modifies it using the user defined wg function, and returns the StepInstance.

`gen_step_instance` is called by `add` method of TraceContext.